### PR TITLE
block a backup process until synchronization ends

### PR DESCRIPTION
**Pull request by lkowalczyk87**
Jira: [CP-1800]

block a backup process until synchronization ends

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-1800]: https://appnroll.atlassian.net/browse/CP-1800?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/backup/services/sync-backup-create.service.ts [hunk 1]
```diff
@@ -142,20 +140,25 @@ export class SyncBackupCreateService {
...
   ): Promise<ResultObject<GetBackupDeviceStatusResponseBody>> {
     const result = await this.getBackupDeviceStatus({
       id,
     })
 
-    if (!result.ok || result.data?.state === BackupState.Error) {
+    if (
+      !result.ok ||
+      result.data?.state === BackupState.Error ||
+      iteration > 60
```
Comment (patryk-sierzega) [inline packages/app/src/backup/services/sync-backup-create.service.ts:153]: why 60? Does it mean something?

File: packages/app/src/data-sync/reducers/data-sync.reducer.ts [hunk 2]
```diff
@@ -41,12 +43,14 @@ export const dataSyncReducer = createReducer<DataSyncState>(
         return {
           ...state,
           state: SynchronizationState.Loading,
+          synchronizationProcess: SynchronizationProcessState.InProgress,
```
Comment (patryk-sierzega) [inline packages/app/src/data-sync/reducers/data-sync.reducer.ts:46]: I wonder if this additional field is needed, cause SynchronizationProcessState.InProgress is only when SynchronizationState.Loading, in other cases synchronizationProcess is SynchronizationProcessState.Done? Can't we calculate synchronizationProcess out of state?

File: packages/app/src/backup/services/sync-backup-create.service.ts (hunk 1 above)
Comment (dkarski) [reply packages/app/src/backup/services/sync-backup-create.service.ts:153]: I think that will be nice to provide a CONST variable to describe what it is.

File: packages/app/src/overview/components/backup/backup.component.tsx [hunk 3]
```diff
@@ -57,12 +57,14 @@ interface Props {
   lastBackupDate?: Date
   onBackupCreate: () => void
   onBackupRestore?: () => void
+  isBackupActionDisabled: boolean
```
Comment (dkarski) [inline packages/app/src/overview/components/backup/backup.component.tsx:60]: 🟢 
In the Mudita Center project we don’t use `is` prefix for non-functional values. I add this topic to discuss at the front-end meeting.

File: packages/app/src/data-sync/reducers/data-sync.reducer.ts (hunk 2 above)
Comment (lkowalczyk87) [reply packages/app/src/data-sync/reducers/data-sync.reducer.ts:46]: I wanted to use `state`. Unfortunately, there is a path that does not fit this case :/

File: packages/app/src/overview/components/backup/backup.component.tsx (hunk 3 above)
Comment (lkowalczyk87) [reply packages/app/src/overview/components/backup/backup.component.tsx]: Resolved in 42e6390e76b22a3456b1ce5b1efdbf9da9d4e2e8

File: packages/app/src/backup/services/sync-backup-create.service.ts (hunk 1 above)
Comment (lkowalczyk87) [reply packages/app/src/backup/services/sync-backup-create.service.ts]: Good catch. Resolved in 42e6390e76b22a3456b1ce5b1efdbf9da9d4e2e8

**Review by dkarski — APPROVED** [review]

**Review by patryk-sierzega — APPROVED** [review]
LGTM :)

**Pull request merged**